bin/middlewares/map_style.js

```diff
@@ -30,25 +36,16 @@ module.exports = function(config, languages) {
 
   const builtStyle = styleBuilder(qwantStyle, options);
 
-  return function(req, res) {
-    /* get lang routine */
-    const langParameter = req.query.lang;
-    let matchedLanguage = languages.defaultLanguage;
-    if (langParameter) {
-      matchedLanguage = languages.supportedLanguages.find(supportedLanguage => {
-        return supportedLanguage.code === langParameter;
-      });
-      if (!matchedLanguage) {
-        res.status(400);
-        res.send({ error: `Unsupported language code : ${langParameter}` });
-        return;
-      }
-    } else {
-      matchedLanguage = languages.defaultLanguage;
+  const styleCache = {};
+
+  const getStyle = function({ host, lang, nopoi }) {
+    const cacheKey = `${host};${lang};${nopoi}`;
```

Comment:
I wondered if we had an already imported memoization util like for example [lodash.memoize](https://www.npmjs.com/package/lodash.memoize), to have a pure fonction and avoid writing the cache logic ourselves.
But apparently not and it's simple enough, so :+1: